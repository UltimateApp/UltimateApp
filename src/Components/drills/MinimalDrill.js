import React from 'react';
import { StyleSheet, View } from 'react-native';
import { DrillTypes } from '../../Fixtures/config';

import FitnessDrillIllustration from './FitnessDrillIllustration';
import FrisbeeDrillIllustration from './FrisbeeDrillIllustration';

const MinimalDrill = props => {
  const { drill } = props;

  return (
    <>
      <View style={styles.animation}>
        {drill.type === DrillTypes.FRISBEE && <FrisbeeDrillIllustration drill={drill} minimal />}
        {drill.type === DrillTypes.FITNESS && <FitnessDrillIllustration drill={drill} minimal />}
      </View>
    </>
  );
};

export default MinimalDrill;

const styles = StyleSheet.create({
  animation: {
    flex: 3,
  },
});
