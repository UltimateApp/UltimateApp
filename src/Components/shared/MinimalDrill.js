import React from 'react';
import { StyleSheet, View } from 'react-native';

import DrillIllustration from '../DrillIllustration';

const MinimalDrill = props => {
  const { drill } = props;
  return (
    <>
      <View style={styles.animation}>
        <DrillIllustration drill={drill} minimal />
      </View>
    </>
  );
};

export default MinimalDrill;

const styles = StyleSheet.create({
  animation: {
    flex: 3,
  },
});
