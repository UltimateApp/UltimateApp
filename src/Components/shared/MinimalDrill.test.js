import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, cleanup } from 'react-native-testing-library';
import { createDrill } from '../../Fixtures/TestFixtures';

import MinimalDrill from './MinimalDrill';

afterEach(cleanup);

describe('<MinimalDrill />', () => {
  it('renders correctly', () => {
    const drill = createDrill();
    const tree = renderer.create(<MinimalDrill drill={drill} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
