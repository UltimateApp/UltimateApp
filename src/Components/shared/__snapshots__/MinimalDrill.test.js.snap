// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MinimalDrill /> renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 3,
    }
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onSwipeLeft={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 250,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <RNCWebView
          cacheEnabled={true}
          javaScriptEnabled={true}
          messagingEnabled={false}
          onContentProcessDidTerminate={[Function]}
          onHttpError={[Function]}
          onLoadingError={[Function]}
          onLoadingFinish={[Function]}
          onLoadingProgress={[Function]}
          onLoadingStart={[Function]}
          onMessage={[Function]}
          onShouldStartLoadWithRequest={[Function]}
          source={
            Object {
              "uri": "https://www.youtube.com/embed/JkVHrA5o23o",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              Object {
                "backgroundColor": "#ffffff",
              },
              undefined,
            ]
          }
          useSharedProcessPool={true}
        />
      </View>
    </View>
  </View>
</View>
`;
